### ISSUE TITLE
Documentation improvement: it's not so clear about which VM ecosystem utils are included in each deploy option

### ISSUE THREAD

**Issue by lucianolacurcia** [VictoriaMetrics/VictoriaMetrics#2158]
**Is your feature request related to a problem? Please describe.**
It's not so clear in the docs which utils are included in vmsingle helm chart. For instance, as I see it, the vmagent is included but the vmalert is not included. is this correct?

**Describe the solution you'd like**
Just a Venn Diagram showing what utils and components from the VictoriaMetrics ecosystem are included in each way of deploy or install victoriametrics. (helms, operator, binaries, etc)

**Additional context**
For example at my first try I was trying to deploy vmsingle using a helm chart, I also deployed the vmagent for pulling metrics, but then I realized that apparently vmagent is included in the deployment of vmsingle using the chart (and the prometheus.yaml file can be added to the config file as parameter). I also deployed vmalert for notifying AlertManager because its not included

To sum up, I suggest to clarify which components are included in each way of deploying victoriametrics

**Comment by YuriKravetc** [comment]
Hi,

Thanks for pointing it out.
We will try to make it more clear in the documentation.

As for the list of charts included, it can be found here:
https://github.com/VictoriaMetrics/helm-charts

Best regards,

**Comment by YuriKravetc** [comment]
VMSingle can perform a self scrape, so pretty much includes vmagent functionality.

**Comment by valyala** [comment]
> Ok, so the scrape capabilities of VMSingle are not intended to use as metrics scraping from other sources than VMSingle itself?

Single-node VictoriaMetrics can scrape metrics from external Prometheus targets - see [these docs](https://docs.victoriametrics.com/#how-to-scrape-prometheus-exporters-such-as-node-exporter) . I'm unsure whether this functionality is supported in the [helm chart](https://github.com/VictoriaMetrics/helm-charts) and [operator](https://docs.victoriametrics.com/operator/) though. @YuriKravetc , could you figure out this and update docs accordingly?

**Comment by valyala** [comment]
FYI, the docs for [VictoriaMetrics operator](https://docs.victoriametrics.com/operator/) have been improved over time. Also docs for [VictoriaMetrics helm charts](https://victoriametrics.github.io/helm-charts/) have been also improved over time.

**Comment by hasanawad94** [comment]
@valyala @lucianolacurcia I'm also a bit lost in the documentation about what VMsignle is exactly. Can you please provide more information on the differences of deploying VMsingle and VMagent with VMcluster ?

**Comment by jiekun** [comment]
Hi. I think VMSingle is a CRD defined here: https://docs.victoriametrics.com/operator/resources/vmsingle/
It refer to the single node version of VictoriaMetrics: https://docs.victoriametrics.com/single-server-victoriametrics/

And for the difference between VictoriaMetrics Single Node and VictoriaMetrics Cluster, please see: https://docs.victoriametrics.com/faq/#which-victoriametrics-type-is-recommended-for-use-in-production---single-node-or-cluster

**Comment by hasanawad94** [comment]
> Hi @hasanawad94 . I think VMSingle is a CRD defined here: https://docs.victoriametrics.com/operator/resources/vmsingle/ It refer to the single node version of VictoriaMetrics: https://docs.victoriametrics.com/single-server-victoriametrics/
> 
> And for the difference between VictoriaMetrics Single Node and VictoriaMetrics Cluster, please see: https://docs.victoriametrics.com/faq/#which-victoriametrics-type-is-recommended-for-use-in-production---single-node-or-cluster

Thanks for your reply  @jiekun ! Just one thing to confirm based on the documentation here : 
![image](https://github.com/user-attachments/assets/00850f5f-243d-4677-96a8-e9effa8ac7b1)
I can safely assume that VMsingle crd is mainly composed of vmagent, and vmcluster components and other components from victoriametrics ?

**Comment by jiekun** [comment]
@hasanawad94 No. VMSingle is a single component (please consider it as a process) that capable of scraping & receive metrics data, store them, and provide query API.

There are many other components from VictoriaMetrics ecosystem such as vmalert, vmauth, vmbackup and more.

Please check the link I provided above, which introduced the difference between VictoriaMetrics Single Node and VictoriaMetrics Cluster.

**Comment by hasanawad94** [comment]
@jiekun still confused with the documentation I went to https://docs.victoriametrics.com/single-server-victoriametrics/#high-availability
Which is under  single node version of VictoriaMetrics to understand high availability and deduplication and in both sections vmagent is mentioned. 
This is why I want to understand if vmagent is within the VMsingle CRD so I can know which documentation to rely on when I want to configure it.

**Comment by hasanawad94** [comment]
> @hasanawad94 No. vmagent is a standalone component for scraping metrics and remote writing to VictoriaMetrics Single Node or VictoriaMetrics Cluster. In the meantime, VictoriaMetrics Single Node can also scrape metrics. They are different components. For HA in VictoriaMetrics Single Node, you can config multiple vmagent to scrape metrics, and remote write to multiple VictoriaMetrics Single Node. Those VMSingle instances are considered mirror to each other.

What's the differences between deploying VMagent to scrape and write to VictoriaMetrics Single Node or VMcluster and to just deploy VictoriaMetrics Single Node(VMsingle) to rely on it to scrape and act a storage ?

**Comment by jiekun** [comment]
It's described in https://docs.victoriametrics.com/vmagent/#motivation.

Also, VictoriaMetrics Single Node is not designed for horizontal scaling. For example, if you want have 10 scrape targets and you want to scale up to support 100k scrape targets:
- In VictoriaMetrics Single Node you may need to add resources to the node it deployed.
- With vmagent, you can distributed many of them to different nodes, and config to them share the scrape jobs of 100k targets.

Please read: 
- https://docs.victoriametrics.com/single-server-victoriametrics/#capacity-planning
- https://docs.victoriametrics.com/single-server-victoriametrics/#scalability-and-cluster-version
